File: backend/RaceRoHub/src/main/java/com/racero/hub/model/Track.java
package com.racero.hub.model;

import jakarta.persistence.Entity;
import jakarta.persistence.GeneratedValue;
import jakarta.persistence.GenerationType;
import jakarta.persistence.Id;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

@Entity
@Getter
@Setter
@NoArgsConstructor
public class Track {
    @GeneratedValue(strategy = GenerationType.AUTO)
    @Id
    public Long id;

    public String location;
    public String distance;
    public String description;
    public String imageUrl;

}

File: frontend/racerohub/lib/models/track.dart
class Track {
  final int id;
  final String location;
  final String distance;
  final String description;
  final String imageUrl;

  const Track({
    required this.id,
    required this.location,
    required this.distance,
    required this.description,
    required this.imageUrl,
  });

  factory Track.fromJson(Map<String, dynamic> json) => Track(
    id: (json['id'] as num).toInt(),
    location: (json['location'] ?? '') as String,
    distance: (json['distance'] ?? '') as String,
    description: (json['description'] ?? '') as String,
    imageUrl: (json['imageUrl'] ?? '') as String,
  );
}
